File: golang/1207.go
func uniqueOccurrences(arr []int) bool {
	m := make(map[int]int)

	for _, v := range arr {
		m[v]++
	}

	n := make(map[int]struct{})
	for _, v := range m {
		if _, ok := n[v]; ok {
			return false
		}
		n[v] = struct{}{}
	}
	return true
}

File: rust/1207.rs
impl Solution {
    pub fn unique_occurrences(arr: Vec<i32>) -> bool {
        use std::collections::HashMap;
        use std::collections::HashSet;

        let mut m = HashMap::new();
        let mut res = HashSet::new();

        for num in arr.iter() {
            *m.entry(num).or_insert(0) += 1
        }

        for (_, count) in &m {
            res.insert(count);
        } 

        m.len() == res.len()
    }
}
